spec/02-integration/02-cmd/03-reload_spec.lua

```diff
@@ -81,9 +82,78 @@ describe("kong reload", function()
 
   describe("errors", function()
     it("complains about missing PID if not already running", function()
+      helpers.prepare_prefix()
+
       local ok, err = helpers.kong_exec("reload --prefix " .. helpers.test_conf.prefix)
       assert.False(ok)
       assert.matches("Error: nginx not running in prefix: " .. helpers.test_conf.prefix, err, nil, true)
     end)
   end)
+
+  it("clears the 'kong' shm", function()
+      local client
+
+      assert(helpers.start_kong())
+
+      finally(function()
+        helpers.stop_kong()
+        if client then
+          client:close()
+        end
+      end)
+
+      client = helpers.admin_client()
+      local res = assert(client:get("/"))
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      local prng_seeds_1 = json.prng_seeds
+      client:close()
+
+      assert(helpers.kong_exec("reload --prefix " .. helpers.test_conf.prefix))
+      ngx.sleep(1)
+
+      client = helpers.admin_client()
+      local res = assert(client:get("/"))
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      local prng_seeds_2 = json.prng_seeds
+      client:close()
+
+      for k in pairs(prng_seeds_1) do
+        assert.is_nil(prng_seeds_2[k])
+      end
+  end)
```

Comment:
Need to handle `database = off` in this test case (https://travis-ci.org/Kong/kong/jobs/530544583).